src/aiodynamo/credentials.py

```diff
@@ -158,8 +158,9 @@ def __init__(
         self,
         *,
         path: Optional[Path] = None,
-        profile_name: str = "default",
+        profile_name: Optional[str] = None,
     ) -> None:
+        profile_name = profile_name or os.environ.get("AWS_PROFILE", "default")
```

Comment:
if you assign to a different variable name here in replace the `or` with an `if`, you don't need the `str(...)` in line 178.

Reply:
actually, this seems like a mypy [bug](https://github.com/python/mypy/issues/14338) so maybe just using `typing.cast` here and removing the `str(...)` below would be fine too.

Reply:
For a quick fix:

`pro = pro if pro else od.env(...)`

or possibly more legible / correct:

```py
if profile_name is None:
    profile_name = os.env(...)
```

Reply:
you cannot narrow an existing name in mypy... at least not that way https://mypy-play.net/?mypy=latest&python=3.11&gist=774404f9e0075cf360b4b7e54b013546

Reply:
Right, good point. I should have remembered that.
`x if x else y` it is then.

Reply:
how is mypy okay with that but not `or` :( 

anyway, inlined looks good to me if it pacifies mypy

Reply:
Added `typing.cast` to take care of that. I initially wrote in as an inline `x if x is not None else y` but when formatting was applied it looked hideous so I went with a multi-line `if` statement.